client/internal/packagessyncer.go

```diff
@@ -98,7 +99,7 @@ func (s *packagesSyncer) initStatuses() error {
 }
 
 // doSync performs the actual syncing process.
-func (s *packagesSyncer) doSync(ctx context.Context) {
+func (s *packagesSyncer) doSync(ctx context.Context, mutex *sync.Mutex) {
 	hash, err := s.localState.AllPackagesHash()
```

Comment:
Why isn't the mutex locked right from the top of doSync? `AllPackagesHash` is also susceptible to the race. It reads data that is written by `SetAllPackagesHash` below.

Reply:
Ah, Missed that. I actually thought that it only handles logging. Will change the mutex. Thanks for pointing it out